LanguageFeatures/Patterns/type_inference_A19_t02.dart

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Calculate the static type of the pattern. Using that value type,
+/// recurse through the pattern again downwards to the leaf subpatterns filling
+/// in any missing types in the pattern. This process may also insert implicit
+/// coercions and casts from dynamic when values flow into a pattern during
+/// matching
+///
+/// @description Check that the calculation of the static type of a record
+/// pattern performs implicit coercions
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=patterns,records
+
+import "../../Utils/expect.dart";
+import "../../Utils/static_type_helper.dart";
+
+main() {
+  var (double x1, n: double x2) = (42, n: 42);
```

Comment:
This is again phase 1 causing coercions in the initializing expression during phase 2. We should be able to use an initializing expression which is a variable, such that '42' isn't coerced into '42.0' before we even construct the record.